edgelet/iotedge/src/support_bundle.rs

```diff
@@ -313,6 +382,89 @@ where
+        let mut inspect = ShellCommand::new("docker");
+
+        /***
+         * Note: just like inspect, this assumes using windows containers on a windows machine.
+         */
+        #[cfg(windows)]
+        inspect.args(&["-H", "npipe:////./pipe/iotedge_moby_engine"]);
+
+        inspect.args(&["network", "ls"]);
+        inspect.args(&["--format", "{{.Name}}"]);
+        let inspect = inspect.output();
+
+        let result = if let Ok(result) = inspect {
+            if result.status.success() {
+                String::from_utf8_lossy(&result.stdout).to_string()
+            } else {
+                println!(
+                    "Could not find network names: {}",
+                    String::from_utf8_lossy(&result.stderr)
+                );
+                "azure-iot-edge".to_owned()
+            }
+        } else {
+            println!("Could not find network names: {}", inspect.err().unwrap());
+            "azure-iot-edge".to_owned()
+        };
+
+        Ok((result.lines().map(String::from).collect(), state))
+    }
+
+    fn write_docker_network_to_file(
+        mut state: BundleState<M>,
+        network_name: String,
+    ) -> Result<BundleState<M>, Error> {
+        state.print_verbose(&format!(
+            "Running docker network inspect for {}",
+            network_name
+        ));
+        let mut inspect = ShellCommand::new("docker");
+
+        /***
+         * Note: just like inspect, this assumes using windows containers on a windows machine.
+         */
+        #[cfg(windows)]
+        inspect.args(&["-H", "npipe:////./pipe/iotedge_moby_engine"]);
+
+        inspect.args(&["network", "inspect", &network_name, "-v"]);
+        let inspect = inspect.output();
+
+        let (file_name, output) = if let Ok(result) = inspect {
+            if result.status.success() {
+                (format!("network/{}.json", network_name), result.stdout)
+            } else {
+                (format!("network/{}_err.json", network_name), result.stderr)
+            }
+        } else {
+            let err_message = inspect.err().unwrap().description().to_owned();
+            println!(
+                "Could not reach docker. Including error in bundle.\nError message: {}",
+                err_message
+            );
+            (
+                format!("network/{}_err_docker.txt", network_name),
+                err_message.as_bytes().to_vec(),
+            )
+        };
+
+        state
+            .zip_writer
+            .start_file_from_path(&Path::new(&file_name), state.file_options)
+            .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle)))?;
+
+        state
+            .zip_writer
+            .write(&output)
+            .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle)))?;
+
+        state.print_verbose(&format!("Got docker network inspect for {}", network_name));
+        drop(network_name);
```

Comment:
Why do you need the `drop` ?

Remove it from `write_inspect_to_file` too.

Reply:
A while ago clippy complained about not consuming the String. It doesn't now, so I'll remove it